metricbeat/internal/metrics/opt.go

```diff
@@ -40,6 +42,27 @@ func OptUintWith(i uint64) OptUint {
 	}
 }
 
+// None marks the Uint as not having a value.
+func (opt *OptUint) None() {
+	opt.exists = false
+}
+
+// Exists returns true if the underlying value is valid
+func (opt OptUint) Exists() bool {
+	return opt.exists
+}
+
+// IsNone returns true if the value exists
+func (opt OptUint) IsNone(i uint64) {
+	opt.value = i
+	opt.exists = true
+}
+
+// Value returns true if the value exists
+func (opt OptUint) Value() (uint64, bool) {
```

Comment:
again, do we need to return a tuple? How about just returning the value and 0 if it is not set?

Reply:
Not a strong opinion, but returning a tuple forces to check if the value is valid, I find this less error prone, though it also forces to have more verbose code.